otx/api/usecases/exportable_code/demo/demo_package/executors/sync_pipeline.py

```diff
@@ -83,12 +84,14 @@ def run(self, input_stream: Union[int, str], loop: bool = False) -> None:
 
         for frame in streamer:
             # getting result for single image
+            start_time = time.perf_counter()
             annotation_scene = self.single_run(frame)
             output = self.visualizer.draw(frame, annotation_scene, {})
             self.visualizer.show(output)
             if self.visualizer.output:
                 saved_frames.append(frame)
             if self.visualizer.is_quit():
                 break
-
+            # visualize video not faster then 30 FRS
```

Comment:
A user reported that during inference on fast machines, the visualization speed was faster than the original video speed. So to prevent it, I put a 30 FPS threshold for visualization, because that's the usual frame frequency. 
But your question made me realize that it's better to check the original VideoStreamer FPS and check not to exceed it.
Thank you